web/src/activity.ts

```diff
@@ -166,6 +189,50 @@ export function send_presence_to_server(redraw?: () => void): void {
     });
 }
 
+export async function fetch_presence_user_setting(): Promise<void> {
```

Comment:
Can you explain why this function and `populate_users()` in the next commit have to be made async?

Reply:
Since I am fetching the presence data using this function, where I call the API endpoint, as you suggested in your previous comments, the API should only be called when the "Users" tab of the settings is used. If I do not make it asynchronous, the data is not displayed correctly on the frontend, as it calculates the `last_active_data` from an empty data structure because the API has not yet fetched the data. I want `populate_users` to wait until the data is fetched and then proceed with rendering the active users table on the frontend.

Therefore, I introduced `async` and used `await` in `populate_users` because `populate_users` is responsible for creating the `active_users_table`. When it is called, it will first fetch the data for the past 10 years or more, and only then display the table on the frontend.